### PR TITLE
GraalVM example for Kotlin AWS SDK

### PR DESCRIPTION
**Pull request by Kashif-E**
This pull request adds a new project `cloudwatch-logs-reader` to the repository. The project is a Kotlin-based application that reads log events from AWS CloudWatch Logs. It uses the AWS SDK for Kotlin and is configured to run as a native image using GraalVM.

### Key Changes:
- Added `cloudwatch-logs-reader` project to `settings.gradle.kts`.
- Implemented the main functionality in `Main.kt` to read log events from CloudWatch Logs.
- Configured Gradle tasks for building and running the application as a native image using GraalVM.
- Added reflection configuration for GraalVM in `reflect-config.json`.
- Updated `ReadMe.md` with detailed instructions on how to use the project, including prerequisites, building, running, and reflection configuration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/cloudwatch-logs-reader/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    binaries.all {
+        resources.autodetect()
+        // Add the META-INF/native-image directory to the classpath for reflection configuration
+        configurationFileDirectories.from(file("src/main/resources/META-INF/native-image"))
+    }
+}
+```
+This configuration ensures that the necessary reflection metadata is available at runtime for the native image.
+For more details, you can refer to the [official GraalVM documentation on reflection configuration](https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection/).
+
```
Comment (lauzadis) [inline examples/cloudwatch-logs-reader/ReadMe.md:79]: This reflection config was not necessary in my local tests. Can you add an explanation about when users would need to configure this?

File: examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        client.getLogEventsPaginated(request).buffer(4).collect { response ->
+            response.events?.forEach { logEvent ->
+                println("Log Event: ${logEvent.message} ${logEvent.timestamp?.formattedDataTime()}")
+            }
+
+        }
+
+    } catch (ex: Exception) {
+        println("Error: ${ex.message}")
```
Comment (lauzadis) [inline examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt:65]: correctness: You should probably just let the exception get thrown. Right now when there is an error the program still exits with code 0, indicating success.

File: examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    } finally {
+        client.close()
+    }
+}
+
+fun Long.formattedDataTime(): String {
+    val timestamp = Instant.fromEpochMilliseconds(this)
+    val dateTime = timestamp.toLocalDateTime(TimeZone.UTC)
+    return "${dateTime.monthNumber}-${dateTime.year} ${dateTime.hour}:${dateTime.minute}:${dateTime.second}"
```
Comment (lauzadis) [inline examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt:74]: nit: missing `dateTime.dayOfMonth`

**Review by lauzadis — COMMENTED** [review]
Thanks for the contribution! It looks mostly good, I have some small suggestions / corrections.

File: examples/build.gradle.kts [hunk 4]
```diff
@@ -15,5 +15,6 @@ allprojects {
     repositories {
         mavenLocal()
         mavenCentral()
+        maven("https://plugins.gradle.org/m2/")
```
[the remark this answers is not in the document]
Comment (Kashif-E) [reply examples/build.gradle.kts:18]: I added this to test an older version of GraalVM plugin but  can be removed as its no longer required.

File: examples/cloudwatch-logs-reader/ReadMe.md (hunk 1 above)
Comment (Kashif-E) [reply examples/cloudwatch-logs-reader/ReadMe.md:79]: The reflection config is there as a placeholder, its required only when there's  a library that uses mongo included in the project such as mongo kotlin client.

Comment (lauzadis) [reply examples/cloudwatch-logs-reader/ReadMe.md:79]: That sounds good, can you add that summary to the README? I don't want users to think configuring the reflection is required.

File: examples/cloudwatch-logs-reader/ReadMe.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+## CloudWatch Logs Reader
```
[the remark this answers is not in the document]
Comment (ianbotsf) [reply examples/cloudwatch-logs-reader/ReadMe.md:1]: I think I disagree. The end value demonstrated to the reader here is both GraalVM _and_ reading logs from CloudWatch. I think this name/header is in line with our other service-specific samples.

File: examples/cloudwatch-logs-reader/ReadMe.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Kotlin and is configured to run as a native image using GraalVM.
+
+### How to Use
+
+#### Prerequisites
+
+<ul>
+<li> JDK 17</li>
+<li> GraalVM</li>
+<li> Gradle</li>
```
Comment (ianbotsf) [inline examples/cloudwatch-logs-reader/ReadMe.md]: **Nit**: Is a standalone installation of Gradle required? I think just downloading/cloning the examples and running a Gradle wrapper (**gradlew** or **gradlew.bat**) should suffice...

File: examples/cloudwatch-logs-reader/ReadMe.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+}
+```
+
+This task uses the GraalVM native image plugin to build and run the application with the specified arguments.
+
+### Reflection Configuration for GraalVM
+
+When building native images with GraalVM, you need to configure reflection manually. This is done by providing a reflection configuration file. The configuration file should be placed in the `src/main/resources/META-INF/native-image` directory.
+
+Here is an example of a reflection configuration file (`reflect-config.json`):
```
Comment (ianbotsf) [inline examples/cloudwatch-logs-reader/ReadMe.md]: **Nit**: Link to the **reflect-config.json** file included in this example

File: examples/cloudwatch-logs-reader/ReadMe.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lauzadis) [reply examples/cloudwatch-logs-reader/ReadMe.md]: I think we should at least mention GraalVM _somewhere_ in the title to improve discoverability. What do you think of `cloudwatch-logs-graalvm`?

File: examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+ * 3. Uses the `CloudWatchLogsReader` to fetch and print log events from the specified log stream.
+ * 4. Handles any exceptions that occur during the process and prints error messages.
+ * 5. Closes the `CloudWatchLogsClient` after processing.
+ *
+ * Example:
+ * tasks.named<org.graalvm.buildtools.gradle.tasks.NativeRunTask>("nativeRun") {
+ * this.runtimeArgs = listOf("us-southeast-1", "my-log-group", "my-log-stream")
+ * }
+ *
+ * Ensure that the AWS credentials and necessary permissions are configured properly for accessing CloudWatch Logs. */
```
Comment (ianbotsf) [inline examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt:34]: **Comment**: Nice KDocs! 😊

File: examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        val request = GetLogEventsRequest {
+            this.logGroupName = logGroupName
+            this.logStreamName = logStreamName
+        }
+        client.getLogEventsPaginated(request).buffer(4).collect { response ->
+            response.events?.forEach { logEvent ->
+                println("Log Event: ${logEvent.message} ${logEvent.timestamp?.formattedDataTime()}")
+            }
+
+        }
```
Comment (ianbotsf) [inline examples/cloudwatch-logs-reader/src/main/kotlin/Main.kt]: **Suggestion** (optional): Many paginators in the SDK have a convenience extension method for flat-mapping from the page down to individual items. You could simplify this by using the `.events()` method here:

```kotlin
client
    .getLogEventsPaginated(request)
    .buffer(4)
    .events()
    .collect { logEvent -> println("Log Event: ${logEvent.message} ${logEvent.timestamp?.formattedDataTime()}") }
}
```

**Review by ianbotsf — COMMENTED** [review]
This'll be a great example to add to our repo! Looks like I reviewed this as you were making changes so feel free to ignore any out-of-date feedback!

**Review by ianbotsf — APPROVED** [review]

File: examples/cloudwatch-logs-reader/README.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
+## CloudWatch Logs Reader with GraalVM Native Image
+
+### Overview
+
+This project is a Kotlin-based application that reads log events from AWS CloudWatch Logs. It uses the AWS SDK for
+Kotlin and is configured to run as a native image using GraalVM. More information about GraalVM native images can be found in the  [official documentation](https://www.graalvm.org/latest/reference-manual/native-image/).
```
Comment (lauzadis) [inline examples/cloudwatch-logs-reader/README.md:6]: nit: unnecessary double space between "the [official..."

File: examples/cloudwatch-logs-reader/README.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  {
+    "name": "com.example.YourClass",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true,
+    "allDeclaredFields": true
+  }
+]
+```
+
+In the build.gradle.kts file, ensure that the META-INF/native-image directory is added to the classpath for reflection configuration:
```
Comment (lauzadis) [inline examples/cloudwatch-logs-reader/README.md:65]: nit: monospace `build.gradle.kts` and `META-INF/native-image`

**Review by lauzadis — APPROVED** [review]

**Comment by Kashif-E** [comment]
Thanks @ianbotsf and @lauzadis for merging 🚀

**Pull request merged**